neuralmagic/nm-vllm
# [ CI ] Fan Out Strategy

SUMMARY:
* update `nm-build-test` workflow to run each test group on separate gpu
* update `nm-test` to receive a specific test directory
* convert `nm-test-whl` action to receive one test directory for test group
* remove concept of skip lists
* remove gptq models (marlin is too flaky)

ADVANTAGES:
* much faster wall clock time for the test
* ability to re-run just the failed jobs for spurious cases
* possible to 

DISADVANTAGES:
* testmo tracking is a bit more complex --- since each test group is no a separate run
* code coverage tracking is now very complex --- since no single run goes over all tests
---> mitigant: we could have a single run for code coverage which runs over all the tests that we run ~weekly

FOLLOW UP PR:
* enable DISTRIBUTED 
* randomly assign various python versions
* update the names of the `TEST` workflow somehow so that in the GH UI we can see more easily which test group failed
